File: backend-repo/FaniverseBE/src/main/java/fantastic/faniverse/community/controller/dto/CommentResponse.java
package fantastic.faniverse.community.controller.dto;

import fantastic.faniverse.community.domain.Comment;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;

import java.time.LocalDateTime;

@Getter
@Builder
@AllArgsConstructor
public class CommentResponse {
    private Long postId;
    private LocalDateTime createdAt;
    private String content;
    public static CommentResponse convertCommentResponse(Comment comment) {
        return CommentResponse.builder()
                .postId(comment.getPost().getId())
                .content(comment.getContent())
                .createdAt(comment.getCreatedAt())
                .build();
    }
}

File: frontend-repo/FaniverseFE2/app/src/main/java/com/example/login/model/CommentResponse.kt
package com.example.login.model

import java.time.LocalDateTime

data class CommentResponse(
    val content: String,
    val createdAt: LocalDateTime
) {
    companion object {
        fun fromEntity(comment: CommentResponse): CommentResponse {
            return CommentResponse(
                content = comment.content,
                createdAt = comment.createdAt
            )
        }
    }
}
